core/domain/blog_statistics_domain.py

```diff
@@ -0,0 +1,1165 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects relating to blog post statistics."""
+
+from __future__ import annotations
+
+import datetime
+
+from core import utils
+
+from typing import Dict, TypedDict, Union, overload
+
+# This is same as base_models.ID_Length.
+BLOG_POST_ID_LENGTH = 12
+
+
+@overload
+def _repack_views_stats(
+    stats: BlogPostViewsAggregatedStats,
+) -> None: ...
+
+
+@overload
+def _repack_views_stats(
+    stats: AuthorBlogPostViewsAggregatedStats,
+) -> None: ...
+
+
+def _repack_views_stats(
+    stats: Union[
+        BlogPostViewsAggregatedStats, AuthorBlogPostViewsAggregatedStats
+    ]
+) -> None:
+    """Repacks aggregated stats domain object to contain necessary keys removing
+    older stats. We will only maintain views_by_hour for past 3 days
+    (including the ongoing day) and views_by_date keyed to 3 months and delete
+    the rest for these fields.
+
+    Args:
+        stats: BlogPostViewsAggregatedStats|AuthorBlogPostViewsAggregatedStats.
+            The stats domain object for which hourly views are to be generated.
+    """
+    stats.views_by_date = {
+        k: stats.views_by_date[k] for k in list(stats.views_by_date)[:3]
+    }
+    stats.views_by_hour = {
+        k: stats.views_by_hour[k] for k in list(stats.views_by_hour)[:3]
+    }
+
+
+@overload
+def _repack_reads_stats(
+    stats: BlogPostReadsAggregatedStats,
+) -> None: ...
+
+
+@overload
+def _repack_reads_stats(
+    stats: AuthorBlogPostReadsAggregatedStats,
+) -> None: ...
+
```

Comment:
Ditto as above, what error MyPy is throwing? without overloading!

Reply:
Dittoo as above